components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppShortcutManager.kt

```diff
@@ -119,3 +156,40 @@ class WebAppShortcutManager(
         storage.removeManifests(startUrls)
     }
 }
+
+/**
+ * Creates custom version of [BrowserIcons] for loading web app icons.
+ *
+ * This version has its own cache to avoid affecting tab icons.
+ */
+@Suppress("LongMethod")
+fun webAppIcons(
+    context: Context,
+    httpClient: Client
+) = BrowserIcons(
+    context = context,
+    httpClient = httpClient,
+    generator = DefaultIconGenerator(cornerRadiusDimen = null),
+    preparers = listOf(
+        TippyTopIconPreparer(context.assets),
+        MemoryIconPreparer(pwaIconMemoryCache),
+        DiskIconPreparer(pwaIconDiskCache)
+    ),
+    loaders = listOf(
+        MemoryIconLoader(pwaIconMemoryCache),
+        DiskIconLoader(pwaIconDiskCache),
+        HttpIconLoader(httpClient),
+        DataUriIconLoader()
+    ),
+    decoders = listOf(
+        AndroidIconDecoder(ignoreSize = true),
+        ICOIconDecoder()
+    ),
+    processors = listOf(
+        MemoryIconProcessor(pwaIconMemoryCache),
+        ResizingProcessor(),
+        DiskIconProcessor(pwaIconDiskCache),
+        ColorProcessor(),
+        AdaptiveIconProcessor()
```

Comment:
I think in a follow-up we should look at the storage of the icons - I feel like for installed apps we want to explicitly store the icon and not rely on a cache.

Reply:
Filed #3799